client/layout/masterbar/item.jsx

```diff
@@ -36,6 +41,15 @@ class MasterbarItem extends Component {
 		}
 	};
 
+	shouldSkipRender = compareProps( { ignore: [ 'onClick' ] } );
+	shouldComponentUpdate( nextProps ) {
+		return ! this.shouldSkipRender( this.props, nextProps );
+	}
+
+	onClick = e => {
+		return this.props.onClick( e );
+	};
```

Comment:
I need some help understanding how this works. Is the goal to prevent `MasterbarItem` from rerendering when the parent component passes a new function instance as `onClick` on every parent rerender?

If yes, then why do we need the `onClick` wrapper method? We can continue passing `this.props.onClick` to the `<a>` element and nothing changes, does it?

I see that the `compareProps` API is not terribly good when using it to implement `shouldComponentUpdate`. It was originally created for React-Redux `connect` custom comparators like `areStatePropsEqual`. I wish we could write an one-liner like:
```js
shouldComponentUpdate: compareHelper( { ignore: [ 'onClick' ] } );
```

Reply:
Thanks for taking a look at the PR, @jsnajdr!

Yes, that's right. During a simple navigation, it looks like the handler instance changes several times, and that's causing unnecessary re-renders. Part of the validation I'm looking for is whether you think this is the right approach (taking care of it inside the component), or whether something else should be done from the outside (e.g. look into memoization where needed).

We need the `onClick` wrapper method to perform a late bind on the handler. If we simply use `this.props.onClick`, the component will be bound to the particular handler that was passed to it, and since we're skipping later renders, we'll be stuck with a stale handler. If we bind to the component instance's `onClick` wrapper, however, we'll always redirect to the current handler in `this.props`.

As for your one-liner, that's totally doable! Want me to write a little wrapper around `compareProps` that works the way you suggested, and place it in `lib`?